app/controllers/application_controller.rb

```diff
@@ -608,12 +620,27 @@ def should_check_authorization?
   end
 
   def validate_headers
-    if response.headers.has_key?('Content-Length') && response.headers['Content-Length'].to_i < 0
+    if response.headers.key?('Content-Length') && response.headers['Content-Length'].to_i.negative?
       raise CustomErrors::BadHeaderError
     end
-    if response.headers.has_key?(:content_length) && response.headers[:content_length].to_i < 0
+    if response.headers.key?(:content_length) && response.headers[:content_length].to_i.negative?
       raise CustomErrors::BadHeaderError
     end
   end
 
+  def sanitize_notes(json)
+    if json.nil?
+      nil
+    elsif json.is_a? Hash
+      json
+    elsif json.is_a? String
+      begin
+        JSON.parse(json)
+      rescue JSON::ParserError => _e
+        { 'comment' => json }
+      end
+    else
+      raise CustomErrors::BadRequestError, 'Invalid notes input.'
+    end
+  end
```

Comment:
your last input case:

```
notes: [1,2,3] => notes: [1,2,3] [WRONG?]
```

won't pass validation here

Reply:
Not sure what you mean here. That input case works

Reply:
(although functionality has now changed so ill resolve this)